palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/repository/K8sBackingStore.java

```diff
@@ -0,0 +1,240 @@
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+
+public class K8sBackingStore implements BackingStore {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(K8sBackingStore.class);
+
+    private final Config config = new ConfigBuilder().build();
+    private final KubernetesClient client = new DefaultKubernetesClient(config);
+
+    private static final String NAMESPACE = "default";
+    private final String namespace = Optional.ofNullable(client.getNamespace()).orElse(NAMESPACE);
+
+    public K8sBackingStore() {
+        LOGGER.info("Current namespace is: {}", this.namespace);
+
+        try {
+            this.client.namespaces().list().getItems().forEach(ns -> LOGGER.info("Found namespace {} with status: {}", ns.getMetadata().getName(), ns.getStatus()));
+        } catch (KubernetesClientException e) {
+            LOGGER.info("Not running in kubernetes or RBAC not entitled");
+        }
+    }
+
+    /**
+     * Clean up the resource
+     */
+
+    public void close() {
+        client.close();
+    }
+
+    public String convertKeyToCompatible(final String key) throws
+            IllegalArgumentException {
+        //only allow lower case alphanumberic character or -
+        if (key.trim().length() == 0) {
+            throw new IllegalArgumentException();
+        }
+        return key.toLowerCase().replaceAll("[^a-z0-9\\-]", "");
+    }
+
+
+    /**
+     * Store the given data in the backing store. The byte array <code>value</code> is assumed to encode an object of
+     * the type represented by the class <code>valueClass</code>. The <code>key</code> must not be empty or
+     * <code>null</code>. If a time to live duration is required on this entry, then the optional specified should not
+     * be empty. Durations must not be negative.
+     * <p>
+     * It is the responsibility of the backing store to ensure that the <code>valueClass</code> is stored along with the
+     * object byte array. The standard way to do this is to store the type name of the class. See {@link
+     * Class#getTypeName()}.
+     *
+     * @param key        the cache key
+     * @param valueClass the object type represented in the byte array
+     * @param value      the encoded object
+     * @param timeToLive an optional time to live, maybe be empty
+     * @return true if and only if the the cache entry was made successfully
+     * @throws IllegalArgumentException if the duration is negative
+     * @throws IllegalArgumentException if <code>key</code> is empty (once whitespace is trimmed)
+     */
+    @Override
+    public boolean add(final String key, final Class<?> valueClass,
+                       final byte[] value, final Optional<Duration> timeToLive) {
+
+        if (key == null) {
+            throw new IllegalArgumentException("key");
+        }
+        requireNonNull(valueClass, "valueClass");
+        requireNonNull(value, "value");
```

Comment:
null check the timeToLive

Reply:
it's Optional - code below accounts for that

Reply:
The optional field itself can be null which will cause a NPE at line 110.